src/audio/multiband_drc/multiband_drc.c

```diff
@@ -64,20 +64,25 @@ static void multiband_drc_reset_state(struct multiband_drc_state *state)
 		multiband_drc_iir_reset_state_ch(&state->deemphasis[i]);
 }
 
-static int multiband_drc_eq_init_coef_ch(struct sof_eq_iir_biquad *coef,
+static int multiband_drc_eq_init_coef_ch(struct comp_dev *dev, struct sof_eq_iir_biquad *coef,
 					 struct iir_state_df2t *eq)
 {
 	int ret;
 
 	eq->coef = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
 			   sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS);
-	if (!eq->coef)
+	if (!eq->coef) {
+		comp_err(dev, "Failed to allocate EQ coefficients");
 		return -ENOMEM;
+	}
 
 	/* Coefficients of the first biquad and second biquad */
 	ret = memcpy_s(eq->coef, sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS,
 		       coef, sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS);
-	assert(!ret);
```

Comment:
the assert is intentional, if memcpy broke we screwed up bad and should stop.

Reply:
Ok, Reverted the original code